### DISCUSSION TITLE
API to list the schemas and columns of a table of a supabase project

### DISCUSSION THREAD

**Discussion by JonathanPicques** [supabase/supabase#6890] [Questions] [answered]
I want to create the best Supabase integration for my no-code tool, and I wanted to know if there was an API to list the schemas and query the columns of a specific table to be able to present a simple UI to my users?

I would like to be able to create a high-level UI like this:
![image](https://user-images.githubusercontent.com/1839757/169321305-5d7ccf66-adbd-414c-927d-cbf560841bf6.png)
This is a screenshot of our Airtable integration, they provide https://airtable.com/api/meta where you can query the columns of a table

Something like `supabaseMeta.getSchemas(): string[]` and `supabaseMeta.getColumns('schema_name', 'my_table_name'): {name:string,type:string}[]` would be awesome

**Comment by steve-chavez** [comment]
https://github.com/supabase/postgres-meta does contain that info but is not exposed to clients(@soedirgo CMIIW).

**Reply by JonathanPicques** [reply]
> https://github.com/supabase/postgres-meta does contain that info but is not exposed to clients(@soedirgo CMIIW).

Yeah I suppose I should have mentioned I wanted these informations using the project api key, not connecting to the postgres server directly.

https://user-images.githubusercontent.com/1839757/169418742-b36e8c06-26eb-419c-b0a1-9d2c5d7b648a.mp4

**Comment by JonathanPicques** [comment, marked as answer]
notalexander on discord came with a brilliant find: https://github.com/zernonia/supabase-schema
Under the hood they use the API discovery from PostgREST to create a visual schema of your supabase project. It works perfectly for my use-case

**Comment by softmarshmallow** [comment]
Would be nice to have this schema fetching in supabase-js !

**Comment by softmarshmallow** [comment]
Hey @JonathanPicques  Did you also manage to get schema other than public? (and listing all available schemas)

:)

**Reply by JonathanPicques** [reply]
Sadly nothing more than https://github.com/zernonia/supabase-schema allows

**Comment by tjmcdonough** [comment]
If you are just after the public database schema then this comment on another post does the job https://github.com/orgs/supabase/discussions/27603#discussioncomment-9898742
